pkg/master/master_test.go

```diff
@@ -531,54 +527,62 @@ func testInstallThirdPartyAPIListVersion(t *testing.T, version string) {
 		},
 	}
 	for _, test := range tests {
-		_, fakeClient, server, assert := initThirdParty(t, version)
-		defer server.Close()
-
-		if test.items == nil {
-			fakeClient.ExpectNotFoundGet(etcdtest.PathPrefix() + "/ThirdPartyResourceData/company.com/foos/default")
-		} else {
-			setupEtcdList(fakeClient, "/ThirdPartyResourceData/company.com/foos/default", test.items)
-		}
+		func() {
+			_, etcdserver, server, assert := initThirdParty(t, version)
+			defer server.Close()
+			defer etcdserver.Terminate(t)
+
+			client := etcdserver.Client
+			if test.items != nil {
+				setupEtcdList(client, "/ThirdPartyResourceData/company.com/foos/default", test.items)
+			}
 
-		resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
-		if !assert.NoError(err) {
-			return
-		}
-		defer resp.Body.Close()
+			resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
```

Comment:
I'm actually fixing the test here.
What I did is I opaqued the body of for loop into func() {}(), to make deferred functions to really be called at the end of the loop, not to have all of them be run at the end of the test.
So this is WAI.